launcher.py

```diff
@@ -0,0 +1,405 @@
+            self.install_dependency(system)
+
+    def install_dependency(self, system):
+        """
+        Installs the dependency for the specified operating system.
+
+        Args:
+            system (str): The operating system type.
+        """
+        try:
+            commands = self._get_install_commands(system)
+            for cmd in commands:
+                if type(cmd) == str:
+                    run_command(cmd)
+                else:
+                    cmd()
+        except Exception as e:
+            print(f"Error: {e}.\n\nInstall {self.name} manually.")
+            sys.exit(1)
+
+    def _get_check_commands(self, system):
+        if system in self.check_cmd:
+            return self.check_cmd[system]
+        if "all" in self.check_cmd:
+            return self.check_cmd["all"]
+        return [f"{self.package_name.get(system, self.name)} --version"]
+
+    def _get_install_commands(self, system):
+        if system in self.install_cmd:
+            return self.install_cmd[system]
+        if "all" in self.install_cmd:
+            return self.install_cmd["all"]
+        if system not in self._installers:
+            print(f"Unspported sytem {system}. Please Install {self.name} manually.")
+            sys.exit(1)
+
+        return self._installers[system]()
+
+    def _get_install_commands_linux(self):
+        import distro
+
+        distro = distro.id().lower()
+        if distro not in self._linux_distro_map:
+            print(f"Unsupported linux distro {distro}. Please Install {self.name} manually.")
+            sys.exit(1)
+
+        return [f"{self._linux_distro_map[distro]} {{self.package_name.get(system, self.name)}}"]
+
+    def _get_install_commands_darwin(self):
+        return [f"brew install {{self.package_name.get(system, self.name)}}"]
+
+    def _get_install_commands_windows(self):
+        if not check_msys2_installed():
+            install_msys2()
+        return [f"pacman -S --needed --noconfirm {{self.package_name.get(system, self.name)}}"]
+
+
+# pkg-config should come before sndfie and nlohmann-json
+dependencies = [
+    DependencyHandler("cmake", {"Windows": "mingw-w64-x86_64-cmake"}, {"all": ["cmake --version"]}),
+    DependencyHandler(
+        "g++",
+        {"Windows": "mingw-w64-x86_64-gcc", "Darwin": "gcc"},
+        {"all": ["g++ --version"]},
+        {"Windows": ["pacman -S --needed --noconfirm base-devel mingw-w64-x86_64-toolchain"]},
+    ),
+    DependencyHandler("pkg-config"),
+    DependencyHandler("ffmpeg", {"Windows": "mingw-w64-x86_64-ffmpeg"}, {"all": ["ffmpeg -version"]}),
+    DependencyHandler(
+        "libsndfile",
+        {"Windows": "mingw-w64-x86_64-libsndfile", "Linux": "libsndfile1-dev"},
+        {"all": ["pkg-config --exists sndfile"]},
+    ),
+    DependencyHandler(
+        "nlohmann-json",
+        {"Windows": "mingw-w64-x86_64-nlohmann-json", "Linux": "nlohmann-json3-dev"},
+        {"all": ["pkg-config --exists nlohmann_json"]},
+    ),
+]
```

Comment:
I won't push for this and we can come back to it at a later date, but I'd probably go for something like a `dataclass` here instead of just putting them in the global scope. Same for the webbrowser line with the address and port number. Not critical or urgent, but something to keep in mind